### ISSUE TITLE
scan_rr.go should be generated

### ISSUE THREAD

**Issue by miekg** [miekg/dns#793]
All, but `scan_rr`'s per RR functions are generated from the structs in `types.go`. There is not actually a reason why this can't happen for these as well. Would cut back on maintenance and typos in this file.

**Comment by tmthrgd** [comment]
I took a brief shot at this and it's not the easiest thing to do. So far I've got quite a few unfortunate special-cases, because most of the tags aren't about the presentation format. I think this would be worth doing, but it might be an undertaking.

A small tool I developed that might be helpful is [here](https://gist.github.com/tmthrgd/917b25210ebe45e3df75d0f5bbfb6dac). It sorts all the functions and global map keys in a file, which minimises the diff as the functions would move around otherwise. (It should be used after yanking the not to be generated stuff out of the file).
